### PR TITLE
Accept Valgrind options via VG_OPTIONS env var

### PR DESCRIPTION
**Pull request by rafie**
Avoid quoting woes when using --vg-options

### CODE REVIEW

File: RLTest/__main__.py [hunk 1]
```diff
@@ -208,7 +208,7 @@ def do_normal_conn(self, line):
 parser.add_argument(
     '--vg-suppressions', default=None, help='path valgrind suppressions file')
 parser.add_argument(
-    '--vg-options', default='--leak-check=full --errors-for-leak-kinds=definite', dest='vg_options', help='valgrind [options]')
+    '--vg-options', default=None, dest='vg_options', help='valgrind [options]')
```
Comment (MeirShpilraien) [inline RLTest/__main__.py]: Why changing the default?

File: RLTest/__main__.py [hunk 2]
```diff
@@ -309,7 +309,12 @@ def __init__(self):
             if self.args.env.endswith('existing-env'):
                 print(Colors.Bred('can not use valgrind with existing-env'))
                 sys.exit(1)
-            vg_debugger = debuggers.Valgrind(options=self.args.vg_options, suppressions=self.args.vg_suppressions, fail_on_errors=not(self.args.vg_no_fail_on_errors), leakcheck=not(self.args.vg_no_leakcheck)
+            if self.args.vg_options is None:
+                self.args.vg_options = os.getenv('VG_OPTIONS', '--leak-check=full --errors-for-leak-kinds=definite')
```
Comment (MeirShpilraien) [inline RLTest/__main__.py]: Why not just using `--vg-options` when running RLTest?

Comment (rafie) [reply RLTest/__main__.py]: Of course we can: --vg-options will continue to work exactly as before.
But sometimes, within a script, passing multiple options to vg-options gets you into bash quoting problems that make the code messy, hard to understand and maintain.
Using an env var easily resolves this.

Comment (MeirShpilraien) [reply RLTest/__main__.py]: I do not mind also allow env var but lets not change the default, we can have the env var override the parameter and write a warnig. How does it sound?

File: RLTest/__main__.py (hunk 1 above)
Comment (rafie) [reply RLTest/__main__.py]: The command line option always overrides the env var. I've changed the default to None so I can detect whether the command line option was provided, and if not, probe for the env var. If the latter is missing, the old default is taken.

File: RLTest/__main__.py (hunk 2 above)
Comment (MeirShpilraien) [reply RLTest/__main__.py]: Ohh ok, got it.

**Pull request merged**